tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -581,6 +681,86 @@ public boolean equals(Object obj) {
         public int hashCode() {
             return lengths.hashCode() ^ styles.hashCode() ^ usages.hashCode() ^ orders.hashCode();
         }
+
+        @Override
+        public int compareTo(ParameterMatcher other) {
+            return ComparisonChain.start()
+                .compare(lengths, other.lengths, Length.ITERABLE_COMPARE)
+                .compare(styles, other.styles, Style.ITERABLE_COMPARE)
+                .compare(usages, other.usages, Usage.ITERABLE_COMPARE)
+                .compare(orders, other.orders, Order.ITERABLE_COMPARE)
+                .result();
+        }
+
+        public ParameterMatcher merge(ParameterMatcher that) {
+            // if 3 of the 4 rows are equal, merge the other two
+            if (lengths.equals(that.lengths)
+                && styles.equals(that.styles)
+                && usages.equals(that.usages)) {
+                return new ParameterMatcher(lengths, styles, usages, Sets.union(orders, that.orders));
+            } else if (lengths.equals(that.lengths)
+                && styles.equals(that.styles)
+                && orders.equals(that.orders)) {
+                return new ParameterMatcher(lengths, styles, Sets.union(usages, that.usages), orders);
+            } else if (lengths.equals(that.lengths)
+                && usages.equals(that.usages)
+                && orders.equals(that.orders)) {
+                return new ParameterMatcher(lengths, Sets.union(styles, that.styles), usages, orders);
+            } else if (styles.equals(that.styles)
+                && usages.equals(that.usages)
+                && orders.equals(that.orders)) {
+                return new ParameterMatcher(Sets.union(lengths, that.lengths), styles, usages, orders);
+            }
+            return null;
+        }
+
+        public ParameterMatcher slim() {
+            return new ParameterMatcher(
+                lengths.equals(Length.ALL) ? ImmutableSet.of() : lengths,
+                    styles.equals(Style.ALL) ? ImmutableSet.of() : styles,
+                        usages.equals(Usage.ALL) ? ImmutableSet.of() : usages,
+                            orders.equals(Order.ALL) ? ImmutableSet.of() : orders
+                );
+        }
+
+        /**
+         * Compact a set of FormatParameters into a (smaller) collection of ParameterMatchers
+         */
+        public static Set<ParameterMatcher> compact(Collection<FormatParameters> expanded) {
```

Comment:
I think it's worth noting that the order of matchers (whether it's important or not) is not preserved.